src/librustc/middle/stability.rs

```diff
@@ -559,9 +583,19 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         // Deprecated attributes apply in-crate and cross-crate.
         if let Some(id) = id {
             if let Some(depr_entry) = self.lookup_deprecation_entry(def_id) {
+                // If the deprecation is scheduled for a future Rust
+                // version, then we should display no warning message.
+                let deprecated_in_future_version = if let Some(sym) = depr_entry.attr.since {
+                    let since = sym.as_str();
+                    !deprecation_in_effect(since.as_ref())
```

Comment:
Please don't use `as_ref` to get the `&str`; that tends to be a footgun. Passing `&since` should cause deref coercion to pass the `&str` to `deprecation_in_effect`.